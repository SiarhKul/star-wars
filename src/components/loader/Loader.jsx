import React from "react";
import { useSelector } from "react-redux";
import BarLoader from "react-spinners/BarLoader";
import { css } from "@emotion/react";

export const Loader = () => {
	const { isDataLoadedFromServer } = useSelector(state => state.loading);
	const override = css`
		width: 100%;
		height: 5px;
		background-color: red;
	`;
	return (
		<div className="spinner">
			{isDataLoadedFromServer && <BarLoader css={override} />}
		</div>
	);
};
