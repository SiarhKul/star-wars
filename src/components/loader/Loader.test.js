import * as redux from "react-redux";
import { Loader } from "./Loader";
import React from "react";

describe("Test Loader component", () => {
	const loader = () => shallow(<Loader />);
	let component;
	let spyOnUseSelector;

	beforeEach(() => {
		spyOnUseSelector = jest
			.spyOn(redux, "useSelector")
			.mockReturnValue({ isDataLoadedFromServer: false });
		component = loader();
	});

	it("should 'Loader' render ", () => {
		expect(component).toMatchSnapshot();
	});

	it("should 'Loader' render if data have been loaded ", () => {
		expect(component.find("Loader").exists()).toBeFalsy();
	});
	it("should 'Loader' render if data have been loaded ", () => {
		spyOnUseSelector = jest
			.spyOn(redux, "useSelector")
			.mockReturnValue({ isDataLoadedFromServer: true });
		component = loader();
		expect(component.find("Loader").exists()).toBeTruthy();
	});
});
