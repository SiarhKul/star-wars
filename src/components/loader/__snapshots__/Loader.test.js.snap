// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loader component should 'Loader' render  1`] = `
<div
  className="spinner"
/>
`;
